src/main/java/com/redhat/quarkus/prodapp/ProductVersionsResource.java

```diff
@@ -47,6 +47,11 @@ public void updateExtension(@PathParam String version, @PathParam String extensi
 
     @POST
     public void addVersion(Version version) {
+        if (version.getBaseVersion().isEmpty()){
+            double baseVersion = getVersions().stream().mapToDouble(v->Double.parseDouble(v.getId()))
```

Comment:
this should be handled by selecting the newest/default version in `toRealVersion`

Reply:
Ok, but there is no check/validation to ensure that it happens every time we add a new version. If somebody adds a new version without checking `newest`, then it pass a null `Version` object. We can change the above code to a simple `version.setBaseVersion(MockStorage.defaultVersion);`

Reply:
I insist on using toRealVersion here ;) UI can send `newest` when user doesn't provide a version number

Reply:
Okay,good idea !

Reply:
Done !